src/test/java/org/jenkinsci/plugins/workflow/cps/CpsBodyExecutionTest.java

```diff
@@ -214,14 +217,36 @@ public boolean start() throws Exception {
         });
         rr.then(r -> {
             WorkflowRun b = r.jenkins.getItemByFullName("demo", WorkflowJob.class).getBuildByNumber(1);
+            SemaphoreStep.success("wait/1", null);
+            r.assertBuildStatusSuccess(r.waitForCompletion(b));
```

Comment:
Weird diff, but all I did here was removed the redundant call to `SemaphoreStep.waitForStart` and the comment/code regarding "Before the fix for JENKINS-53709" since https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/104 changed the behavior.